data/ignition.template

```diff
@@ -107,6 +107,7 @@ storage:
           [Unit]
           Description=Planner agent quadlet
           Wants=planner-agent-opa.service
```

Comment:
perhaps we want to have "Requires" instead of Wants? it seems we have a strong dependency here.

Reply:
After reviewing the official documentation at the link below, I believe your suggestion to use Requires instead of Wants is a good practice.

However, in both cases, we still encounter error messages. If we change to Requires, the dependency will be stronger, and we'll receive an error message if there's a race condition. Currently, we are still getting errors, likely because the agent is not recognizing that the OPA has started.

Since this line of code was written by someone else before and is not part of my changes, perhaps we should discuss this further. However, you've convinced me that it would be a good practice to switch from Wants to Requires. 

Please let me know if you think we should just go ahead with the change or if it would be better to discuss it in more detail.

full link: https://www.freedesktop.org/software/systemd/man/latest/systemd.unit.html

Reply:
I think if would be nice to change the wants to requires as part of this change. I am not saying to do this instead of the "after" you added, I mean to do it with it.
Obviously the code was written before, we are here to make it better with every commit :)

Reply:
the reason I think it is good to change it to requires is because it expresses the type of dependency we have here to my understanding.

Reply:
wait - actually reading the docs I see:

> Default Dependencies
> Default dependencies are similar to implicit dependencies, but can be turned on and off by setting DefaultDependencies= to yes (the default) and no, while implicit dependencies are always in effect. See section "Default Dependencies" in respective man pages for the effect of enabling DefaultDependencies= in each unit types.
> 
> For example, target units will complement all configured dependencies of type Wants= or Requires= with dependencies of type After=. See [systemd.target(5)](https://www.freedesktop.org/software/systemd/man/latest/systemd.target.html#) for details. Note that this behavior can be opted out by setting DefaultDependencies=no in the specified units, or it can be selectively overridden via an explicit Before= dependency."

does not it mean that "after" does not change anything?
Let's think, how do we check that the fix work?

Reply:
It's a little bit confusing but watch that: 

Regarding before and after: 
Those two settings configure ordering dependencies between units. If unit foo.service contains the setting Before=bar.service and both units are being started, bar.service's start-up is delayed until foo.service has finished starting up. After= is the inverse of Before=, i.e. while Before= ensures that the configured unit is started before the listed unit begins starting up, After= ensures the opposite, that the listed unit is fully started up before the configured unit is started.

**And here is explicitly write that they are independent:** 
here is a note in Wants - Note that requirement dependencies do not influence the order in which services are started or stopped. This has to be configured independently with the After= or Before= options. If unit foo.service pulls in unit bar.service as configured with Wants= and no ordering is configured with After= or Before=, then both units will be started simultaneously and without any delay between them if foo.service is activated.

**and here is another sentence that support the combination of wants and after:** 
Note that those settings are independent of and orthogonal to the requirement dependencies as configured by Requires=, Wants=, Requisite=, or BindsTo=. It is a common pattern to include a unit name in both the After= and Wants= options, in which case the unit listed will be started before the unit that is configured with these options.

Reply:
it seems from the docs that After is added by default when you have Wants/Requires, unless you set DefaultDependencies=false.
I may be wrong of course.

Hmm do we have a way to run it on local? Did we try to use "act"? If it is too complicated, yeh we could merge and see, but it would be nice to test it out. For a proper check - we should run it without the "after" and see that it fails sometimes, and then with and see that it does not.

Reply:
Ok, let's merge this since without this fix it totally fails. Also I run it 5 times and it succeeded. We can improve later.